### ISSUE TITLE
allow aria-haspopup on an ```<a>``` element with role=button

### ISSUE THREAD

**Issue by cptvitamin** [validator/validator#18]
aria-haspopup should be allowed on an `<a>` element when it has a role=button  as in:

`<a href="#" role="button" aria-haspop="true" aria-expanded="false">Button</a>`

You can see the failure here on [this gist](https://validator.nu/?doc=https%3A%2F%2Frawgit.com%2Fcptvitamin%2Fc4e348ee46b7abfa6d43%2Fraw%2Facdcbc385cad4ed97d2f4664fa9be1b8f8048fe5%2Fgistfile1.html).
